server/src/services/media.service.ts

```diff
@@ -204,33 +224,50 @@ export class MediaService extends BaseService {
-    const didExtract = shouldExtract && (await this.mediaRepository.extract(asset.originalPath, extractedPath));
+    const processInvalidImages = process.env.IMMICH_PROCESS_INVALID_IMAGES === 'true';
+    const colorspace = this.isSRGB(asset) ? Colorspace.SRGB : image.colorspace;
+    const decodeOptions = { colorspace, processInvalidImages, size: image.preview.size };
+
+    let fullsizePath: string;
+    let extractedPath: string | undefined;
+    if (mimeTypes.isRaw(asset.originalPath)) {
+      let useExtracted = false;
+      extractedPath = StorageCore.getImagePath(asset, AssetPathType.EXTRACTED, ImageFormat.JPEG);
+      if (image.extractEmbedded) {
+        // try extracting embedded preview from RAW
+        // extracted image from RAW is always in JPEG format, as implied from the `jpgFromRaw` tag name
+        const didExtract = await this.mediaRepository.extract(asset.originalPath, extractedPath);
+        useExtracted = didExtract && (await this.shouldUseExtractedImage(extractedPath, image.preview.size));
+      }
 
-    try {
-      const useExtracted = didExtract && (await this.shouldUseExtractedImage(extractedPath, image.preview.size));
-      const inputPath = useExtracted ? extractedPath : asset.originalPath;
-      const colorspace = this.isSRGB(asset) ? Colorspace.SRGB : image.colorspace;
-      const processInvalidImages = process.env.IMMICH_PROCESS_INVALID_IMAGES === 'true';
-
-      const orientation = useExtracted && asset.exifInfo?.orientation ? Number(asset.exifInfo.orientation) : undefined;
-      const decodeOptions = { colorspace, processInvalidImages, size: image.preview.size, orientation };
-      const { data, info } = await this.mediaRepository.decodeImage(inputPath, decodeOptions);
-
-      const options = { colorspace, processInvalidImages, raw: info };
-      const outputs = await Promise.all([
-        this.mediaRepository.generateThumbnail(data, { ...image.thumbnail, ...options }, thumbnailPath),
-        this.mediaRepository.generateThumbnail(data, { ...image.preview, ...options }, previewPath),
-        this.mediaRepository.generateThumbhash(data, options),
-      ]);
-
-      return { previewPath, thumbnailPath, thumbhash: outputs[2] };
-    } finally {
-      if (didExtract) {
-        await this.storageRepository.unlink(extractedPath);
+      if (useExtracted) {
+        fullsizePath = extractedPath;
+      } else {
+        // did not extract or extracted preview is smaller than target size,
+        // convert a full-sized thumbnail from original instead
+        extractedPath = StorageCore.getImagePath(asset, AssetPathType.EXTRACTED, image.preview.format);
+        // const orientation = asset.exifInfo?.orientation ? Number(asset.exifInfo.orientation) : undefined;
+        await this.mediaRepository.generateThumbnail(
+          asset.originalPath,
+          { ...image.preview, colorspace, processInvalidImages, size: Infinity },
+          extractedPath,
+        );
```

Comment:
I'm not sure about this section in general. I think your idea is to allow a deeper zoom for RAW images even when the extracted image shouldn't be used, but this issue isn't unique to RAW images and affects many other formats. Solving this seems out of scope for the PR.